File: ruby/version2.rb
#!/usr/bin/env ruby
# Version 2: Consume SOAP service + translate to Spanish
# Usage: ruby version2.rb <number>
# Example: ruby version2.rb 10

require 'savon'
require 'google_translate'

# Get number from command line argument
number = ARGV[0] || '10'

# Configure SOAP client
client = Savon.client(
  wsdl: 'https://www.dataaccess.com/webservicesserver/NumberConversion.wso?WSDL',
  log: false
)

# Call the NumberToWords operation
response = client.call(
  :number_to_words,
  message: { 'ubiNum' => number }
)

# Extract the result
result = response.body[:number_to_words_response][:number_to_words_result]

# Translate from English to Spanish
translator = GoogleTranslate.new
translated = translator.translate(result, from: 'en', to: 'es')

puts translated

File: perl/version2.pl
#!/usr/bin/perl
# Version 2: Consume SOAP service + translate to Spanish
# Usage: perl version2.pl <number>
# Example: perl version2.pl 10

use strict;
use warnings;
use SOAP::Lite;
use Lingua::Translate;

# Get number from command line argument
my $number = $ARGV[0] || '10';

# Configure SOAP client
my $wsdl = 'https://www.dataaccess.com/webservicesserver/NumberConversion.wso?WSDL';
my $client = SOAP::Lite->service($wsdl);

# Call the NumberToWords operation
my $result = $client->NumberToWords($number);

# Translate from English to Spanish
my $translator = Lingua::Translate->new(src => 'en', dest => 'es');
my $translated = $translator->translate($result);

print "$translated\n";
